nomad/secure_variables_endpoint.go

```diff
@@ -468,3 +477,32 @@ func (sv *SecureVariables) authValidatePrefix(alloc *structs.Allocation, taskNam
 	}
 	return nil
 }
+
+func (sv *SecureVariables) handleCheckAndSet(path string, idx *uint64, wo structs.WriteRequest) error {
```

Comment:
This has lifted the implementation from `Job.Register`'s `-check-index`, which wasn't a bad approach b/c jobs changes infrequently and the `-check-index` is really about making sure a `Job.Register` has the same state as a previous `Job.Plan`. But for secure variables, users are going to want to use CAS as a primitive for locks and other concurrent access approaches, so we should try to be more strict with concurrent updates. We should push this logic all the way down to the state store, similar to how it was done in Consul: [ex. `kvs.go#L363-L388`](https://github.com/hashicorp/consul/blob/main/agent/consul/state/kvs.go#L363-L388)